### PR TITLE
Type array tweaks

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
- Now it uses `zMove` in `GetTypeBasedZMove`.
- Added Arceus forms to the commented data.

## Feature(s) this PR does NOT handle:
Can't think of any other type-related stuff.

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

**Comment by AlexOn1ine** [comment]
Do maxMove already use gTypesInfo? You've changed it for z-moves but not for them.

File: src/battle_z_move.c [hunk 1]
```diff
@@ -378,13 +378,13 @@ static u16 GetTypeBasedZMove(u16 move, u8 battler)
...
 
-    // Get z move from type
-    if (moveType < TYPE_FIRE)
-        return MOVE_BREAKNECK_BLITZ + moveType;
-    else if (moveType >= TYPE_FAIRY)
-        return MOVE_TWINKLE_TACKLE + (moveType - TYPE_FAIRY);
-    else
-        return MOVE_BREAKNECK_BLITZ + (moveType - 1);
+    if (moveType >= NUMBER_OF_MON_TYPES)
+        moveType = TYPE_MYSTERY;
```
Comment (AlexOn1ine) [inline src/battle_z_move.c:382]: Mystery has no zmove. What will be returned in this case?

**Comment by AsparagusEduardo** [comment]
> Do maxMove already use gTypesInfo? You've changed it for z-moves but not for them.

Yep, they do.

File: src/battle_z_move.c (hunk 1 above)
Comment (AsparagusEduardo) [reply src/battle_z_move.c:382]: Normal, due to the bottom check.

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**